recipes/sail/all/conanfile.py

```diff
@@ -0,0 +1,191 @@
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_avif:
+            self.requires("libavif/0.9.3")
+        if self.options.with_gif:
+            self.requires("giflib/5.2.1")
+        if self.options.with_jpeg2000:
+            self.requires("jasper/2.0.33")
+        if self.options.with_jpeg:
+            self.requires("libjpeg/9d")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.3.0")
+        if self.options.with_webp:
+            self.requires("libwebp/1.2.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        except_codecs = []
+
+        if not self.options.with_avif:
+            except_codecs.append("AVIF")
+        if not self.options.with_gif:
+            except_codecs.append("GIF")
+        if not self.options.with_jpeg2000:
+            except_codecs.append("JPEG2000")
+        if not self.options.with_jpeg:
+            except_codecs.append("JPEG")
+        if not self.options.with_png:
+            except_codecs.append("PNG")
+        if not self.options.with_tiff:
+            except_codecs.append("TIFF")
+        if not self.options.with_webp:
+            except_codecs.append("WEBP")
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["SAIL_BUILD_APPS"]     = False
+        self._cmake.definitions["SAIL_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["SAIL_BUILD_TESTS"]    = False
+        self._cmake.definitions["SAIL_COMBINE_CODECS"] = True
+        self._cmake.definitions["SAIL_EXCEPT_CODECS"]  = ";".join(except_codecs)
+        self._cmake.definitions["SAIL_INSTALL_PDB"]    = False
+        self._cmake.definitions["SAIL_THREAD_SAFE"]    = self.options.thread_safe
+        self._cmake.configure(build_folder=self._build_subfolder)
+
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.txt",       src=self._source_subfolder, dst="licenses")
+        self.copy("LICENSE.INIH.txt",  src=self._source_subfolder, dst="licenses")
+        self.copy("LICENSE.MUNIT.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        # Remove CMake and pkg-config rules
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        # Move icons
+        rename(self, os.path.join(self.package_folder, "share"),
```

Comment:
Is it really needed?

Reply:
yes, `share` is not allowed by our hook

Reply:
Yes, I mean, the real motivation of packaging the share folder. Usually it only contains docs, manual and other useless stuff for a packaged binary.